### DISCUSSION TITLE
Debugging timed-out tests

### DISCUSSION THREAD

**Discussion by simplenotezy** [avajs/ava#2997] [Q&A]
How do I debug timed out tests?

We get below error message from time to time. There seems to be no pattern. If we run test again, it will either not time out, or another test will time out.

We also sometimes get "Rejected promise returned by test", which is equally as hard to debug, since it does not state where in the test/which test that timed out. Maybe it's related?

```
  ✖ Timed out while running tests

  50 tests were pending in test/match-making/match-making.spec.ts

  ◌ match-making › match-making.ts › query potentialDoubbleMatchWithTeammate | Should not return an deactived teammate as a potential DoubbleMatch
  ◌ match-making › match-making.ts › query potentialDoubbleMatchWithTeammate | Should return null if a non-existing datingprofile id is used
  ◌ match-making › match-making.ts › query potentialDoubbleMatchWithTeammate | Should return null if a teammate is not in a potential DoubbleMatch
  ◌ match-making › match-making.ts › query potentialDoubbleMatchWithTeammate | Should return null if teammate only has a match with relevantProfile
  ◌ match-making › match-making.ts › query potentialDoubbleMatchWithTeammate | Should return a teammate if a relevantProfile has not liked queryingProfile
```

**Comment by novemberborn** [comment]
This can be hard to pin down. Running with `--serial` can help, so that the tests aren't interleaving. You should then be able to tell which test completed last, so hopefully it's the next one that caused the timeout. It may also be that the first pending test is the one that caused the timeout, I can't quite recall.

**Reply by simplenotezy** [reply]
Thanks for the feedback. It's a bit problematic, because it's only from time to other they fail. But I will remember the `--serial` flag.
